src/app/fdctl/run/tiles/fd_replay.c

```diff
@@ -0,0 +1,731 @@
+  ulong            poh_out_depth;
+  ulong            poh_out_seq;
+
+  fd_wksp_t * poh_out_mem;
+  ulong       poh_out_chunk0;
+  ulong       poh_out_wmark;
+  ulong       poh_out_chunk;
+
+  // Store tile output defs
+  fd_frag_meta_t * store_out_mcache;
+  ulong *          store_out_sync;
+  ulong            store_out_depth;
+  ulong            store_out_seq;
+
+  fd_wksp_t * store_out_mem;
+  ulong       store_out_chunk0;
+  ulong       store_out_wmark;
+  ulong       store_out_chunk;
+
+  // Stake weights output link defs
+  fd_frag_meta_t * stake_weights_out_mcache;
+  ulong *          stake_weights_out_sync;
+  ulong            stake_weights_out_depth;
+  ulong            stake_weights_out_seq;
+
+  fd_wksp_t * stake_weights_out_mem;
+  ulong       stake_weights_out_chunk0;
+  ulong       stake_weights_out_wmark;
+  ulong       stake_weights_out_chunk;
+
+  long last_stake_weights_push_time;
+
+  fd_acc_mgr_t          acc_mgr[1];
+  uchar *               epoch_ctx_mem;
+  fd_exec_epoch_ctx_t * epoch_ctx;
+  fd_exec_slot_ctx_t *  slot_ctx;
+
+  fd_replay_t *         replay;
+
+  fd_wksp_t  * blockstore_wksp;
+  char const * snapshot;
+  char const * incremental;
+  char const * genesis;
+
+  ulong     curr_slot;
+  ulong     parent_slot;
+  ulong     flags;
+  fd_hash_t blockhash;
+  
+  uchar        tpool_mem[FD_TPOOL_FOOTPRINT( FD_TILE_MAX )] __attribute__( ( aligned( FD_TPOOL_ALIGN ) ) );
+  fd_tpool_t * tpool;
+  ulong        max_workers;
+};
+typedef struct fd_replay_tile_ctx fd_replay_tile_ctx_t;
+
+static fd_valloc_t inner_valloc;
+
+FD_FN_CONST static inline ulong
+scratch_align( void ) {
+  return 128UL;
+}
+
+FD_FN_PURE static inline ulong
+loose_footprint( fd_topo_tile_t const * tile FD_PARAM_UNUSED ) {
+  return 16UL * FD_SHMEM_GIGANTIC_PAGE_SZ;
+}
+
+FD_FN_PURE static inline ulong
+scratch_footprint( fd_topo_tile_t const * tile FD_PARAM_UNUSED ) {
+  ulong l = FD_LAYOUT_INIT;
+  l = FD_LAYOUT_APPEND( l, alignof(fd_replay_tile_ctx_t), sizeof(fd_replay_tile_ctx_t) );
+  return FD_LAYOUT_FINI( l, scratch_align() );
+}
+
+FD_FN_CONST static inline void *
+mux_ctx( void * scratch ) {
+  return (void*)fd_ulong_align_up( (ulong)scratch, alignof(fd_replay_tile_ctx_t) );
+}
+
+static void
+before_frag( void * _ctx,
+             ulong  in_idx      FD_PARAM_UNUSED,
+             ulong  seq         FD_PARAM_UNUSED,
+             ulong  sig         FD_PARAM_UNUSED,
+             int *  opt_filter  FD_PARAM_UNUSED ) {
+  fd_replay_tile_ctx_t * ctx = (fd_replay_tile_ctx_t *)_ctx;
+  (void)ctx;
+}
+
+static void
+during_frag( void * _ctx,
+             ulong  in_idx      FD_PARAM_UNUSED,
+             ulong  seq         FD_PARAM_UNUSED,
+             ulong  sig         FD_PARAM_UNUSED,
+             ulong  chunk,
+             ulong  sz,
+             int *  opt_filter  FD_PARAM_UNUSED ) {
+  fd_replay_tile_ctx_t * ctx = (fd_replay_tile_ctx_t *)_ctx;
+  
+  if( FD_UNLIKELY( chunk<ctx->in_chunk0 || chunk>ctx->in_wmark || sz>8192UL ) ) {
```

Comment:
remove magic number and add explanation for constant.
I believe this is the estimate for max txns in a block as of now (or what we can support anyway)

Reply:
Done